### PR TITLE
ci: add a check for properly wrapped errors

### PR DESCRIPTION
**Pull request by stevekuznetsov**
ci: add a check for properly wrapped errors

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

*: fix errors found by the wrapping checker

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Example output:

```
$ go run github.com/polyfloyd/go-errorlint -errorf ./...
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/reconciler/cluster/controller.go:143:77: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/sharding/apiserver/sharded_storage.go:101:102: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/sharding/apiserver/sharded_storage.go:109:102: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/sharding/apiserver/delegating_storage.go:86:24: type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/sharding/apiserver/delegating_storage.go:257:96: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/sharding/apiserver/state.go:235:111: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/server/server.go:137:58: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/home/stevekuznetsov/code/kcp-dev/kcp/src/github.com/kcp-dev/kcp/pkg/server/server.go:144:53: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
exit status 3
```

File: hack/tools.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ package tools
 // This package imports things required by this repository, to force `go mod` to see them as dependencies
 import (
 	_ "github.com/coreydaley/openshift-goimports"
+	_ "github.com/polyfloyd/go-errorlint"
```
Comment (ncdc) [inline hack/tools.go:25]: Mind using the go-install pattern? https://github.com/kcp-dev/kcp/blob/b89a696c843e8da348c810af94cb2b55d961480c/Makefile

Comment (ncdc) [reply hack/tools.go:25]: Should probably file an issue to do this for openshift-goimports too

Comment (ncdc) [reply hack/tools.go:25]: Filed #256 for openshift-goimports & I'll make that change

**Comment by imjasonh** [comment]
I think this is something we can get golangci-lint to check for us instead of involving another tool

**Comment by imjasonh** [comment]
> I think this is something we can get golangci-lint to check for us instead of involving another tool

https://golangci-lint.run/usage/linters/#errorlint

**Comment by stevekuznetsov** [comment]
@imjasonh ah neat! Those are new features :)

**Comment by stevekuznetsov** [comment]
@imjasonh updated the PR to just turn on that check

**Review by imjasonh — APPROVED** [review]

**Pull request merged**